### DISCUSSION TITLE
Get title without truncate

### DISCUSSION THREAD

**Discussion by pwsandoval** [wezterm/wezterm#3232] [Q&A]
Hello, thanks for the support.

How can I get the title without truncatie when it is too long, i.e. if the current directory is /d/dev/discovering/redis-tutorial this is what i get with

tab.active_pane.title return 18:56:56.295 INFO logging > lua: ..edis-tutorial

In other words, the folder name is truncated and ".." is added.

I know I can use tab.active_pane.current_working_dir and get file:///d/Dev/discovering/redis-tutorial (since i set OSC7 in my .zshrc)

But I want to use tab.active_pane.title because with this I also get the program that is running, be it vim, nano, top, etc. unlike tab.active_pane.current_working_dir which always returns the folder name, even if I are using vim or top.

1. How can I get the title with tab.active_pane.title without truncatie when it is too long?
2. Can you also clarify what exactly that this tab.active_pane.title means, I don't quite understand how it returns a title in the same callback where I define the title (format-tab-title)?

**Comment by jankatins** [comment]
This might be set from outside of wezterm, if you e.g. use oh my zsh via osc stuff: https://wezfurlong.org/wezterm/recipes/passing-data.html#osc-0-1-2-for-setting-the-windowpane-title

and set here: https://github.com/mattjj/my-oh-my-zsh/blob/master/terminal.zsh

Do you use one of if these?

**Reply by pwsandoval** [reply]
Yes, I [use this](https://gist.github.com/pwsandoval/79499fbfde43e0a72410d711a5a92a25) based on this example of [wezterm repo](https://github.com/wez/wezterm/blob/main/assets/shell-integration/wezterm.sh)

In case you didn't understand, what happens is that if I get the full name with **tab.active_pane.current_working_dir** but with **tab.active_pane.title** it is truncated when it is very long.

Thanks for your time.

**Reply by wez** [reply]
* OSC 1 sets the "icon" title for a pane.
* OSC 2 sets the title title for a pane.

When wezterm determines the title for a pane to display it, or return it via lua properties:

* The icon title (OSC 1), if set, takes precedence
* Then the OSC 2 title
* If the pane is a local pane, and the title is "wezterm" (the default title of the terminal state), then the foreground process name, if we can resolve it, is used in place of "wezterm"

The title is completely unrelated to OSC 7, which is used to remember the current working directory.

wezterm does not truncate any of these values.

If you are seeing truncated values, it is because whatever applications are setting the title are truncating them.
